### PR TITLE
Wrapper commands for list and forge tps

### PR DESCRIPTION
**Pull request by huantianad**
This pull request is a WIP, and is not complete.

### Overview
This pull request adds wrapper commands for the in-game commands `/list` and `/forge tps`. 

### Reasoning
`/list` and `/forge tps` are very common commands, often used to check the status of servers. Because of this, it would be nice to have them formatted in a better way, and ensure that no extra console output leaks out when running this command on all servers.

### Implementation
We essentially just run the command in server console, and then wait until we find the output of said command, which we match using regex. After a match, we can extract the relevant data, and format it to our liking.

### TODOs
- [ ] Do the `/forge tps` wrapper, error handling for running on vanilla.
- [ ] Better error handling in general, timeout for waiting for server output.
- [ ] Better formatting/colors.
- [ ] Check how this works in game an in console.

### CODE REVIEW

**Review by Inrixia — CHANGES_REQUESTED** [review]
Looks good, but please change the list command to use the builtin status ping, this will return online players as a object that can easily be consumed.

**Comment by Inrixia** [comment]
Forge tps will ofc require console parsing until @Mixu78 can implement fetching tps information in SpookUtils. Ideally the command would support both, attempting to use SpookUtils and falling back to console parsing

**Comment by Inrixia** [comment]
Also re your method for fetching the wrapper core. I have a helper function for that now which you can find in wrapperHelpers here: https://github.com/Inrixia/serverWrapper.ts/blob/ac6b9c2c6d133b84ec814267375f1f448ffc53e5/wrapperHelpers/src/modul.ts#L29
there is an example for using it in the command module
https://github.com/Inrixia/serverWrapper.ts/blob/ac6b9c2c6d133b84ec814267375f1f448ffc53e5/modules/command/src/index.ts#L54

**Comment by Inrixia** [comment]
Also would be cool to use a discord embed to include player head images and give a more formatted response on players online.

File: modules/mineapi/src/commands/list.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/list.ts:19]: Stroutput automatically formats a string for output on all types, so console/discord/ingame will be handled.

File: modules/mineapi/src/commands/tps.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/tps.ts:89]: There's no point in constructing this outside of the return statement. 
Moving it to there would also remove the need to import the ```MessageEmbedOptions``` type.

File: modules/mineapi/src/commands/tps.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/tps.ts:26]: This error really needs to be more descriptive of what actually went wrong.

File: modules/mineapi/src/commands/tps.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/tps.ts:28]: Again no need to construct a object and then just return it. You can return the object directly.

File: modules/mineapi/src/commands/tps.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/tps.ts:32]: You should use ```Number()``` or ```+``` to convert strings to number instead of ```parseFloat``` unless you know you need ```parseFloat```.

Ideally use ```+```.

Ex:
```ts
tickTime: +x[3];
```

File: modules/mineapi/src/commands/tps.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/tps.ts:44]: This could just return without else if's but its not critical. I can change it later after this is merged.

File: modules/mineapi/src/commands/tps.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/tps.ts:64]: Technically you should setup your promise to listen for the console response before calling serverStdin to avoid the case where the response comes from console before you start listening.

File: modules/mineapi/src/commands/tps.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/tps.ts:68]: Define this outside the function call, no need to redefine it every time its called.

File: modules/mineapi/src/commands/tps.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/tps.ts:86]: Please disable debugging before merging

File: modules/mineapi/src/commands/tps.ts
Comment (Inrixia) [inline modules/mineapi/src/commands/tps.ts:92]: Instead of using a un descriptive name like ```x```, use ```dimension``` or whatever ```x``` represents. Easier to read that way.

**Review by Inrixia — CHANGES_REQUESTED** [review]
Looks good overall, few changes that should be done before merging, see comments.

Thanks again for the help

**Comment by huantianad** [comment]
Putting this here as a reminder to make sure server status works with large amounts of players on a server, and to improve formatting

File: modules/mineapi/src/commands/tps.ts
Comment (huantianad) [reply modules/mineapi/src/commands/tps.ts:32]: Yeah figured that out, gotta get used to typescript

File: modules/mineapi/src/commands/tps.ts
Comment (huantianad) [reply modules/mineapi/src/commands/tps.ts:92]: Fixed!

File: modules/mineapi/src/commands/tps.ts
Comment (huantianad) [reply modules/mineapi/src/commands/tps.ts:68]: Moved it just above the function, that should be better

File: modules/mineapi/src/commands/tps.ts
Comment (huantianad) [reply modules/mineapi/src/commands/tps.ts:64]: I've moved the execution of the command into the Promise, think that'll work better

File: modules/mineapi/src/commands/tps.ts
Comment (huantianad) [reply modules/mineapi/src/commands/tps.ts:44]: Would I use ternary statements instead?

File: modules/mineapi/src/commands/tps.ts
Comment (huantianad) [reply modules/mineapi/src/commands/tps.ts:26]: This should now be more descriptive, forgot to change the placeholder.

File: modules/mineapi/src/commands/tpss.ts
Comment (huantianad) [inline modules/mineapi/src/commands/tpss.ts:27]: Just want to make sure, is it ok if I don't return anything for minecraft or console, since executing the command should return an output in console.

Comment (Inrixia) [reply modules/mineapi/src/commands/tpss.ts:27]: Not returning anything for a specific output type will just result in no output when executing the command. So you dont have to. Afik console/ingame using ```/forge tps``` is better anyway so in this case its fine.

**Review by Inrixia — APPROVED** [review]

**Comment by Inrixia** [comment]
Happy to merge this if your happy. Can answer any further questions about code style in discord.

File: modules/mineapi/src/lib/tps.ts
Comment (huantianad) [inline modules/mineapi/src/lib/tps.ts:47]: Just want to double check that this is something that's OK?

**Comment by huantianad** [comment]
> Happy to merge this if your happy. Can answer any further questions about code style in discord.

Yeah it's basically ready to go, I left a few todos in comments, but none are strictly necessary

**Review by Inrixia — APPROVED** [review]

**Comment by Inrixia** [comment]
Apologies for the delay. Thanks again

**Pull request merged**
